src/measure_theory/lp_space.lean

```diff
@@ -1218,4 +1217,390 @@ by { rw eq, exact continuous_pos_part.comp continuous_neg }
+    rw finset.sum_range_sub, },
+  rw h_rw_sum at hx,
+  have hf_rw : (λ n, f n x) = λ n, f n x - f 0 x + f 0 x, by { ext1 n, abel, },
+  rw hf_rw,
+  exact ⟨l + f 0 x, filter.tendsto.add_const _ hx⟩,
+end
+
+lemma cauchy_limit_ℒp [complete_space E] {f : ℕ → α → E} {p : ℝ}
+  (hf : ∀ n, mem_ℒp (f n) (ennreal.of_real p) μ) (hp1 : 1 ≤ p) {B : ℕ → ℝ≥0∞}
+  (hB : ∑' i, B i < ∞) (h_cau : ∀ (N n m : ℕ), N ≤ n → N ≤ m → snorm' (f n - f m) p μ < B N) :
+  ∃ (f_lim : α → E) (hf_lim_meas : measurable f_lim),
+    ∀ ε, 0 < ε → ε < ∞ → (∃ (N : ℕ), ∀ (n : ℕ), N ≤ n → snorm' (f n - f_lim) p μ < ε) :=
+begin
+  have hp_pos := zero_lt_one.trans_le hp1,
+  obtain ⟨f_lim, h_f_lim_meas, h_lim⟩ : ∃ (f_lim : α → E) (hf_lim_meas : measurable f_lim),
+      ∀ᵐ x ∂μ, filter.tendsto (λ n, f n x) filter.at_top (nhds (f_lim x)),
+    from measurable_limit_of_tendsto_metric_ae (λ n, (hf n).1)
+      (ae_tendsto_of_cauchy_snorm' (λ n, (hf n).1) hp1 hB h_cau),
+  use [f_lim, h_f_lim_meas],
+  have h_snorm_lim_rw : ∀ n, snorm' (f n - f_lim) p μ
+      = (∫⁻ a, filter.at_top.liminf (λ m, (nnnorm (f n a - f m a) : ℝ≥0∞)^p) ∂μ) ^ (1/p),
+    from snorm'_lim_sub hp1 h_lim,
+  simp_rw h_snorm_lim_rw,
+  intros ε hε hε_top,
+  obtain ⟨N, h_⟩ : ∃ N, ∀ n m, N ≤ n → N ≤ m → ∫⁻ a, (nnnorm (f n a - f m a) : ℝ≥0∞)^p ∂μ < ε^p/2,
+  { suffices h_snorm : ∃ (N : ℕ), ∀ n m, N ≤ n → N ≤ m → snorm' (f n - f m) p μ < ε / 2^(1/p),
+    { cases h_snorm with N h_snorm,
+      refine ⟨N, λ n m hn hm, _⟩,
+      specialize h_snorm n m hn hm,
+      rwa [snorm', ←@ennreal.lt_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div,
+        ennreal.div_rpow_of_nonneg _ _ hp_pos.le, ←ennreal.rpow_mul, one_div,
+        inv_mul_cancel hp_pos.ne.symm, ennreal.rpow_one] at h_snorm, },
+    suffices h_B : ∃ (N : ℕ), B N ≤ ε/2^(1/p),
+      from ⟨h_B.some, λ n m hn hm, (h_cau h_B.some n m hn hm).trans_le h_B.some_spec⟩,
+    suffices h_tendsto_zero : ∃ (N : ℕ), ∀ n : ℕ, N ≤ n → B n ≤ ε/2^(1/p),
+      from ⟨h_tendsto_zero.some, h_tendsto_zero.some_spec _ (le_refl _)⟩,
+    exact (ennreal.tendsto_at_top_zero.mp (ennreal.tendsto_at_top_zero_of_tsum_lt_top hB))
+      (ε/2^(1/p)) (ennreal.div_pos_iff.mpr ⟨hε.ne.symm, by simp [hp_pos]⟩), },
+  refine ⟨N, λ n hnN, _⟩,
+  rw [←@ennreal.lt_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div],
+  refine (lintegral_liminf_le' (λ m,
+    ((hf n).1.sub (hf m).1).nnnorm.ennreal_coe.ennreal_rpow_const)).trans_lt _,
+  have hε2 : ε^p/2 < ε^p,
+  { refine ennreal.half_lt_self _ _,
+    { exact (ennreal.rpow_pos_of_nonneg hε hp_pos.le).ne.symm, },
+    { exact ennreal.rpow_ne_top_of_nonneg hp_pos.le (lt_top_iff_ne_top.mp hε_top), }, },
+  refine lt_of_le_of_lt _ hε2,
+  refine filter.liminf_le_of_frequently_le' (filter.frequently_at_top.mpr _),
+  exact λ N1, ⟨max N N1, le_max_right _ _, (h_ n (max N N1) hnN (le_max_left _ _)).le⟩,
+end
+
+lemma mem_ℒp_of_cauchy_limit [hp : fact (1 ≤ p)] {f : ℕ → α → E} (hf : ∀ n, mem_ℒp (f n) p μ)
```

Comment:
The typeclass assumption `[hp : fact (1 ≤ p)]` is only important when you want to state something about the metric space structure of `Lp`, because it makes this assumption available to typeclass inference. Since the lemma does not deal with this metric space structure, it is better to go for a usual assumption `hp : 1 ≤ p`. Same thing in another lemma below.